src/user.ts

```diff
@@ -97,7 +97,8 @@ export class User {
         const idTonomyActiveKey = PrivateKey.from("PVT_K1_2bfGi9rYsXQSXXTvJbDAPhHLQUojjaNLomdm3cEJ1XTzMqUt3V");
 
         // TODO need to remove sha256 from this.salt
-        const res = await idContract.newperson(usernameHash.toString(), passwordKey.toString(), this.storage.salt.toString(), createSigner(idTonomyActiveKey));
+        const salt = await this.storage.salt;
```

Comment:
on the member variable `storage`, is it possible to add a typescript type e.g.

```ts
type UserStorage = {
  salt: string
  ...
}
...
export class User {
    keyManager: KeyManager;
    storage: UserStorage
    
    constructor(_keyManager: KeyManager, _storage: PersistantStorage) {
        this.keyManager = _keyManager;
        this.storage = _storage as UserStorage;
    }
```

Reply:
this would make it more visible to expect what exists on the storage member variable

Reply:
i added type to the storage the only bad thing for that is that the user wont be able to add own data in the storage without getting type errors